### PR TITLE
updates to make max http url length and the max header size configura…

### PR DESCRIPTION
**Pull request by sameer7**
…ble. Also exposed isSSL in http request interface

Signed-off-by: Patel, Sameer <sameer.us@gmail.com>

### CODE REVIEW

File: src/main/java/io/vertx/core/http/HttpServerRequest.java [hunk 1]
```diff
@@ -71,6 +71,11 @@
    * @return the HTTP method for the request.
    */
   HttpMethod method();
+  
+  /**
+   * @return whether the HTTP connection is secure
```
Comment (vietj) [inline src/main/java/io/vertx/core/http/HttpServerRequest.java]: please use the same wording that in NetSocket : `true if this {@link io.vertx.core.net.NetSocket} is encrypted via SSL/TLS.`

**Comment by vietj** [comment]
please add tests:

-  that checks the line is truncated at 4000 with default option and one test that shows it is not truncated with a different options
- same for max headers
- add isSsl == true in existing Https tests

**Comment by sameer7** [comment]
I could not find the Https tests for isSSL, can you point me to that. I got the rest.

**Comment by vietj** [comment]
I mean a check on isSSL in HttpTest#testTLS server request handler.

Perhaps we should have the same on websocket too.

**Comment by vietj** [comment]
I see no tests for the maxInitialLineLength and maxHeadersLength feature.

**Comment by sameer7** [comment]
It's there. Line #630 & 726 in my second commit.

**Comment by vietj** [comment]
these tests are for option. we need to have tests with an HttpClient that submit requests with such lines and headers and a verification on the server that the line and headers are correct.

**Comment by sameer7** [comment]
Added tests with HttpClient with server side verification.

**Comment by sameer7** [comment]
Any updates on this?

**Comment by vietj** [comment]
it was not included in 3.2 by lack of time and review. It should be merged in master after review.

**Comment by cescoffier** [comment]
The branch is not mergeable anymore. Could you update ?

**Comment by sameer7** [comment]
Updated.

**Comment by sameer7** [comment]
Did that work for you?

**Comment by vietj** [comment]
thanks for the contribution!

**Comment by sameer7** [comment]
thank you all

**Pull request merged**